x/foundation/msgs.go

```diff
@@ -353,3 +353,28 @@ func (m MsgRevoke) GetSigners() []sdk.AccAddress {
 	signer := sdk.MustAccAddressFromBech32(m.Operator)
 	return []sdk.AccAddress{signer}
 }
+
+var _ sdk.Msg = (*MsgOneTimeMint)(nil)
+
+// ValidateBasic implements Msg.
+func (m MsgOneTimeMint) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(m.Operator); err != nil {
+		return sdkerrors.ErrInvalidAddress.Wrapf("invalid operator address: %s", m.Operator)
+	}
+
+	if m.Amount.Empty() {
+		return sdkerrors.ErrInvalidCoins.Wrapf("The one-time-mint request amount is empty.")
+	}
+
+	if !m.Amount.IsValid() || !m.Amount.IsAllPositive() {
+		return sdkerrors.ErrInvalidCoins.Wrap(m.Amount.String())
```

Comment:
You may add the unit test for it.

Reply:
IsValid func is including IsAllPositive() check. So, I deleted IsAllPositive() check and added unit test for that check.